test/hotspot/jtreg/serviceability/jvmti/GetOwnedMonitorInfo/GetOwnedMonitorInfoTest.java

```diff
@@ -46,19 +46,37 @@ public class GetOwnedMonitorInfoTest {
         }
     }
 
+    private static native void jniMonitorEnter(Object obj);
     private static native int check();
     private static native boolean hasEventPosted();
 
+    private static void jniMonitorEnterAndLetObjectDie() {
+        // The monitor iterator used GetOwnedMonitorInfo used to
+        // assert when an owned monitor with a dead object was found.
+        // Inject this situation into this test that performs other
+        // GetOwnedMonitorInfo testing.
+        Object obj = new Object() { public String toString() {return "";} };
+        jniMonitorEnter(obj);
```

Comment:
I would add a check for `Thread.holdsLock(obj);` after this just to be sure it worked.

Reply:
Done